applications/FemToDemApplication/custom_processes/assign_pressure_id_process.cpp

```diff
@@ -53,9 +57,10 @@ void AssignPressureIdProcess::AssignPressureIdToNodes(
     )
 {
     // Loop over nodes of that submodel to assign prressure id
-    for (auto it = mr_model_part.GetSubModelPart(rSubModelPartName).NodesBegin(); 
-      it != mr_model_part.GetSubModelPart(rSubModelPartName).NodesEnd(); ++it) {
-		(*it).SetValue(PRESSURE_ID, PressureId);
+    #pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(mrModelPart.GetSubModelPart(rSubModelPartName).Nodes().size()); i++) {
+        auto it_node = mrModelPart.GetSubModelPart(rSubModelPartName).NodesBegin() + i;

```

Comment:
Move mrModelPart.GetSubModelPart(rSubModelPartName).NodesBegin() outside the loop, you are accesing to a databse all the time

Reply:
good point